### ISSUE TITLE
Not able to open Financial, Promotion and User Analysis tab

### ISSUE THREAD

**Issue by pikender** [vinsol-spree-contrib/spree-admin-insights#15]
**MySQL Version**

```
pikender@Pikenders-MacBook-Pro:~/spree/store$ mysql --version
mysql  Ver 14.14 Distrib 5.7.10, for osx10.11 (x86_64) using  EditLine wrapper
```

**StackTrace**

*Financial Analysis*

```
Sequel::DatabaseError - Mysql2::Error: Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 't1.number' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by:
  sequel (4.32.0) lib/sequel/core.rb:124:in `convert_exception_class'
  sequel (4.32.0) lib/sequel/database/misc.rb:427:in `raise_error'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:100:in `rescue in _execute'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:78:in `_execute'
  sequel (4.32.0) lib/sequel/adapters/shared/mysql_prepared_statements.rb:36:in `block in execute'
  sequel (4.32.0) lib/sequel/database/connecting.rb:251:in `block in synchronize'
  sequel (4.32.0) lib/sequel/connection_pool/threaded.rb:105:in `hold'
  sequel (4.32.0) lib/sequel/database/connecting.rb:251:in `synchronize'
  sequel (4.32.0) lib/sequel/adapters/shared/mysql_prepared_statements.rb:36:in `execute'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:952:in `execute'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:197:in `execute'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:156:in `fetch_rows'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:141:in `each'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:47:in `block in all'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:892:in `_all'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:47:in `all'
   () Users/pikender/.rvm/gems/ruby-2.3.1@spree/bundler/gems/spree-admin-insights-3f54e56ce69d/app/reports/spree/payment_method_transactions_report.rb:35:in `generate'
   () Users/pikender/.rvm/gems/ruby-2.3.1@spree/bundler/gems/spree-admin-insights-3f54e56ce69d/app/services/spree/report_generation_service.rb:23:in `generate_report'
   () Users/pikender/.rvm/gems/ruby-2.3.1@spree/bundler/gems/spree-admin-insights-3f54e56ce69d/app/controllers/spree/admin/insights_controller.rb:15:in `show'
```

*Promotion Analysis*

```
Sequel::DatabaseError - Mysql2::Error: Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 't1.number' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by:
  sequel (4.32.0) lib/sequel/core.rb:124:in `convert_exception_class'
  sequel (4.32.0) lib/sequel/database/misc.rb:427:in `raise_error'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:100:in `rescue in _execute'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:78:in `_execute'
  sequel (4.32.0) lib/sequel/adapters/shared/mysql_prepared_statements.rb:36:in `block in execute'
  sequel (4.32.0) lib/sequel/database/connecting.rb:251:in `block in synchronize'
  sequel (4.32.0) lib/sequel/connection_pool/threaded.rb:105:in `hold'
  sequel (4.32.0) lib/sequel/database/connecting.rb:251:in `synchronize'
  sequel (4.32.0) lib/sequel/adapters/shared/mysql_prepared_statements.rb:36:in `execute'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:952:in `execute'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:197:in `execute'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:156:in `fetch_rows'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:141:in `each'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:47:in `block in all'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:892:in `_all'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:47:in `all'
   () Users/pikender/.rvm/gems/ruby-2.3.1@spree/bundler/gems/spree-admin-insights-3f54e56ce69d/app/reports/spree/promotional_cost_report.rb:50:in `generate'
   () Users/pikender/.rvm/gems/ruby-2.3.1@spree/bundler/gems/spree-admin-insights-3f54e56ce69d/app/services/spree/report_generation_service.rb:23:in `generate_report'
   () Users/pikender/.rvm/gems/ruby-2.3.1@spree/bundler/gems/spree-admin-insights-3f54e56ce69d/app/controllers/spree/admin/insights_controller.rb:15:in `show'
```

*User Analysis*

```
Sequel::DatabaseError - Mysql2::Error: Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 't1.number' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by:
  sequel (4.32.0) lib/sequel/core.rb:124:in `convert_exception_class'
  sequel (4.32.0) lib/sequel/database/misc.rb:427:in `raise_error'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:100:in `rescue in _execute'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:78:in `_execute'
  sequel (4.32.0) lib/sequel/adapters/shared/mysql_prepared_statements.rb:36:in `block in execute'
  sequel (4.32.0) lib/sequel/database/connecting.rb:251:in `block in synchronize'
  sequel (4.32.0) lib/sequel/connection_pool/threaded.rb:105:in `hold'
  sequel (4.32.0) lib/sequel/database/connecting.rb:251:in `synchronize'
  sequel (4.32.0) lib/sequel/adapters/shared/mysql_prepared_statements.rb:36:in `execute'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:952:in `execute'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:197:in `execute'
  sequel (4.32.0) lib/sequel/adapters/mysql2.rb:156:in `fetch_rows'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:141:in `each'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:47:in `block in all'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:892:in `_all'
  sequel (4.32.0) lib/sequel/dataset/actions.rb:47:in `all'
   () Users/pikender/.rvm/gems/ruby-2.3.1@spree/bundler/gems/spree-admin-insights-3f54e56ce69d/app/reports/spree/user_pool_report.rb:71:in `generate'
   () Users/pikender/.rvm/gems/ruby-2.3.1@spree/bundler/gems/spree-admin-insights-3f54e56ce69d/app/services/spree/report_generation_service.rb:23:in `generate_report'
   () Users/pikender/.rvm/gems/ruby-2.3.1@spree/bundler/gems/spree-admin-insights-3f54e56ce69d/app/controllers/spree/admin/insights_controller.rb:15:in `show'
```

**Useful Info**

- http://stackoverflow.com/questions/34115174/error-related-to-only-full-group-by-when-executing-a-query-in-mysql
- https://dev.mysql.com/doc/refman/5.7/en/group-by-handling.html
  - Check `ANY_VALUE()`, it might be useful

**Comment by zhengpd** [comment]
Same problem here. Solved it by `mysql > SET GLOBAL sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''));` , answer from  http://stackoverflow.com/a/36033983

**Comment by pikender** [comment]
Fixed by #27
